src/game/BoardState.java

```diff
@@ -11,44 +21,142 @@
+		this.pieces.put(new Coordinate(3, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(4, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(5, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(6, 1), new Rabbit(Owner.Player1));
+		this.pieces.put(new Coordinate(7, 1), new Rabbit(Owner.Player1));
+
+		this.pieces.put(new Coordinate(0, 7), new Cat(Owner.Player2));
+		this.pieces.put(new Coordinate(1, 7), new Dog(Owner.Player2));
+		this.pieces.put(new Coordinate(2, 7), new Horse(Owner.Player2));
+		this.pieces.put(new Coordinate(3, 7), new Camel(Owner.Player2));
+		this.pieces.put(new Coordinate(4, 7), new Elephant(Owner.Player2));
+		this.pieces.put(new Coordinate(5, 7), new Horse(Owner.Player2));
+		this.pieces.put(new Coordinate(6, 7), new Dog(Owner.Player2));
+		this.pieces.put(new Coordinate(7, 7), new Cat(Owner.Player2));
+
+		this.pieces.put(new Coordinate(0, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(1, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(2, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(3, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(4, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(5, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(6, 6), new Rabbit(Owner.Player2));
+		this.pieces.put(new Coordinate(7, 6), new Rabbit(Owner.Player2));
+	}
+
+	// maybe make this private?
+	public BoardState(HashMap<Coordinate, AbstractPiece> pieces) {
+		this.pieces = pieces;
+	}
 
-	public BoardState(char[][] map, int turnNumber) {
-		for(int i=0; i<8; i++){
-			for(int k=0; k<8; k++){
-				this.boardArray[i][k] = map[i][k];
+	// converter
+	@Deprecated
+	public BoardState(char[][] chars, int turns) {
+		System.out.println();
+		System.out.println("****************");
+		System.out.println();
+		int randId = new Random().nextInt(1000);
+		System.out.println(
+				"HashMap<Coordinate, AbstractPiece> p" + randId + " = new HashMap<Coordinate, AbstractPiece>();");
+		for (int y = 0; y < 8; y++) {
+			for (int x = 0; x < 8; x++) {
+				char piece = chars[7-y][x];
+				if (piece != ' ') {
+					String player = Character.isUpperCase(piece) ? "Owner.Player1" : "Owner.Player2";
+					piece = Character.toLowerCase(piece);
+					String clazz = "";
+					switch (piece) {
+					case 'r':
+						clazz = "Rabbit";
+						break;
+					case 'k':
+						clazz = "Cat";
+						break;
+					case 'd':
+						clazz = "Dog";
+						break;
+					case 'h':
+						clazz = "Horse";
+						break;
+					case 'c':
+						clazz = "Camel";
+						break;
+					case 'e':
+						clazz = "Elephant";
+						break;
+					}
+					System.out.println("p" + randId + ".put(new Coordinate(" + x + ", " + y + "), new " + clazz + "("
+							+ player + "));");
+				}
 			}
+			throw new RuntimeException("using the wrong constuctor");
 		}
-		this.turnNumber = turnNumber;
 	}
 
+	@Deprecated
```

Comment:
this no longer works, so its now deprecated to help us find instances of its use